kpops/components/base_components/kubernetes_app.py

```diff
@@ -21,19 +27,86 @@ class Config(CamelCaseConfig):
 
 
 # TODO: label and annotations
-
-
 class KubernetesApp(PipelineComponent):
     """Base kubernetes app"""
 
     _type = "kubernetes-app"
     app: KubernetesAppConfig
 
+    _helm_wrapper: Helm | None = None
+    _helm_diff: HelmDiff | None = None
+
     def __init__(self, **kwargs):
         super().__init__(**kwargs)
-        self.check_compatible_name()
+        self.__check_compatible_name()
+
+    @property
+    def helm_wrapper(self) -> Helm:
+        if self._helm_wrapper is None:
+            self._helm_wrapper = Helm(self.config.helm_config)
+        return self._helm_wrapper
+
+    @property
+    def helm_diff(self) -> HelmDiff:
+        if self._helm_diff is None:
+            self._helm_diff = HelmDiff(self.config.helm_diff_config)
+        return self._helm_diff
+
+    @property
+    def helm_release_name(self) -> str:
+        """The name for the Helm release. Can be overridden."""
+        return self.name
+
+    @property
+    def namespace(self) -> str:
+        return self.app.namespace
+
+    def deploy(self, dry_run: bool) -> None:
+        stdout = self.helm_wrapper.upgrade_install(
+            release_name=self.helm_release_name,
+            chart=self.get_helm_chart(),
+            dry_run=dry_run,
+            namespace=self.namespace,
+            values=self.to_helm_values(),
+            flags=HelmUpgradeInstallFlags(version=self.get_helm_repo_config().version),
```

Comment:
Our custom component does not have a repo config (we install a local chart). So we should instead extract a method get_version which returns None by default and can be overridden

Reply:
You can override the `get_helm_repo_config` and set the version to `None` there.

Reply:
Yeah but no repo is needed in our case. And it has some required fields (name/url) which I need to fill

Reply:
Probably version should part ways with the repo config. They are unrelated in how one uses helm